src/v0/destinations/optimizely_fullstack/transform.js

```diff
@@ -0,0 +1,91 @@
+const { EventType } = require('../../../constants');
+const {
+  defaultRequestConfig,
+  simpleProcessRouterDest,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+} = require('../../util');
+const { validateConfig } = require('./util');
+const { CONFIG_CATEGORIES, MAPPING_CONFIG, getTrackEndPoint } = require('./config');
+const { TransformationError, InstrumentationError } = require('../../util/errorTypes');
+
+const responseBuilder = (payload, endpoint, destination) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    const { sdkKey } = destination.Config;
+    response.endpoint = endpoint;
+    response.headers = {
+      'Content-Type': 'application/json',
+      'X-Optimizely-SDK-Key': sdkKey,
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    return response;
+  }
+  // fail-safety for developer error
+  throw new TransformationError('Something went wrong while constructing the payload');
+};
+
+// ref:- https://docs.developers.optimizely.com/experimentation/v3.1.0-full-stack/reference/trackevent
+const trackResponseBuilder = (message, destination) => {
+  const { event, anonymousId } = message;
+  if (!event) {
+    throw new InstrumentationError('Event name is required');
+  }
+  const { baseUrl, trackKnownUsers } = destination.Config;
+  const endpoint = getTrackEndPoint(baseUrl, event);
+  const { name } = CONFIG_CATEGORIES.TRACK;
+  const payload = constructPayload(message, MAPPING_CONFIG[name]);
+  if (!trackKnownUsers) {
+    payload.userId = anonymousId;
```

Comment:
Isn't the userId automatically populated with either of userId or anonId as in config.json we are making sourceFromGeneric to true?

Reply:
logic updated in commit [91af385](https://github.com/rudderlabs/rudder-transformer/pull/1805/commits/91af38557e68179a96c3c94de1a44c90c7ed4ef2)